vignettes/irecist.Rmd

```diff
@@ -0,0 +1,877 @@
+  set_values_to = exprs(AVALC = "NON-iCR/NON-iUPD")
+)
+
+
+ibor_icpd <- event_joined(
+  description = paste(
+    "Define confirmed progressive disease (iCPD) for best overall response (iBOR) as",
+    "iUPD followed by iCPD with only other iUPD and NE responses in between"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  first_cond_upper = AVALC.join == "iCPD",
+  condition = AVALC == "iUPD" &
+    all(AVALC.join %in% c("iCPD", "iUPD", "NE")),
+  set_values_to = exprs(AVALC = "iCPD")
+)
+
+
+ibor_iupd <- event(
+  description = "Define unconfirmed progressive disease (iUPD) for best overall response (iBOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "iUPD",
+  set_values_to = exprs(AVALC = "iUPD")
+)
+
+ibor_ne <- event(
+  description = paste(
+    "Define not evaluable (NE) for best overall response (iBOR) as iSD, NON-iCR/NON-iUPD,",
+    "or NE (should be specified after ibor_isd and ibor_non_icriupd)"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC %in% c("iSD", "NON-iCR/NON-iUPD", "NE"),
+  set_values_to = exprs(AVALC = "NE")
+)
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    tmp_event_nr_var = event_nr,
+    order = exprs(event_nr,ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(ibor_icr, ibor_ipr, ibor_isd, ibor_non_icriupd, ibor_iupd, ibor_icpd, ibor_ne,  no_data_missing),
+    set_values_to = exprs(
+      PARAMCD = "IBOR",
+      PARAM = "Immune Best Overall Response by Investigator (confirmation not required)",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "iRECIST",
+      AVAL = aval_resp_new(AVALC),
+      ANL01FL = "Y"
+    )
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, AVALC, ADT, RANDDT, ANL01FL),
+  filter = PARAMCD == "IBOR"
+)
+```
+
+## Derive Response Parameters requiring Confirmation {#confirm}
+
+Any of the above response parameters can be repeated for "confirmed" responses
+only. For these the function `admiral::derive_extreme_event()` can be used with
+different events. Some of the other functions from above can then be re-used
+passing in these confirmed response records. See the examples below of derived
+parameters requiring confirmation. The assessment and the confirmatory
+assessment here need to occur at least 28 days apart *(without any +1 applied to
+this calculation of days between visits)*, using the `icrsp_y_cr`,
+`icrsp_y_ipr`, `icbor_icr`, and `icbor_ipr` event. Here the confirmation period
+and the `keep_source_vars` argument is updated, as well as the `first_cond_upper` and 
+`condition` for the iRECIST values.
+
+```{r }
+confirmation_period <- 28
+
+icrsp_y_icr <- event_joined(
+  description = paste(
+    "Define confirmed response as iCR followed by iCR at least 28 days later and",
```

Comment:
in description parameter also use cofirmation_period instead of 28

Reply:
updated